### DISCUSSION TITLE
earliestRestrictedContributionDate field of ContributionsCollection object is not working

### DISCUSSION THREAD

**Discussion by fanny** [community/community#24427] [Apps, API and Webhooks]
<p>The <a href="https://developer.github.com/v4/object/contributionscollection/#earliestrestrictedcontributiondate" rel="nofollow noopener">earliestRestrictedContributionDate</a> field of ContributionsCollection object is not working, even if user has enabled private contribution counts.</p>
<p>as my case:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/26193c506688ed12a81cbf279ef138f00f9883e7.png" data-download-href="/uploads/short-url/5r2d4ItcbZif9URfwM3bA4t1jMP.png?dl=1" title="Captura de tela_2019-05-20_22-02-43.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/14113480/181084704-d6bc1a4f-cbf8-4840-b2d8-7097eab8907f.png" alt="Captura de tela_2019-05-20_22-02-43.png" data-base62-sha1="5r2d4ItcbZif9URfwM3bA4t1jMP" width="690" height="298" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26193c506688ed12a81cbf279ef138f00f9883e7_2_690x298.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26193c506688ed12a81cbf279ef138f00f9883e7_2_1035x447.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/26193c506688ed12a81cbf279ef138f00f9883e7.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26193c506688ed12a81cbf279ef138f00f9883e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Captura de tela_2019-05-20_22-02-43.png</span><span class="informations">1366×591 83.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>when i run the query:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/172f3f48c9252fb2c1528844cd4e75b5e08cdd21.png" data-download-href="/uploads/short-url/3j6cVqJc2vbbJm3J43VOSiKdj4B.png?dl=1" title="Captura de tela_2019-05-20_22-07-24.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/14113480/181084705-64199928-798e-4ff6-8592-7c77c1a5ad8c.png" alt="Captura de tela_2019-05-20_22-07-24.png" data-base62-sha1="3j6cVqJc2vbbJm3J43VOSiKdj4B" width="690" height="331" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/172f3f48c9252fb2c1528844cd4e75b5e08cdd21_2_690x331.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/172f3f48c9252fb2c1528844cd4e75b5e08cdd21_2_1035x496.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/1/172f3f48c9252fb2c1528844cd4e75b5e08cdd21.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/172f3f48c9252fb2c1528844cd4e75b5e08cdd21_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Captura de tela_2019-05-20_22-07-24.png</span><span class="informations">1322×636 65.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I get null, this field is deprecated?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>This field is working as intended. If I execute this query:</p>
<pre><code>query getFirstContribution {
  user(login: "lee-dohm") {
    login
    contributionsCollection {
      startedAt
      endedAt
      contributionYears
      earliestRestrictedContributionDate
      hasAnyRestrictedContributions
      hasAnyContributions
      restrictedContributionsCount
    }
  }
}
</code></pre>
<p>Then I receive:</p>
<pre><code>{
  "data": {
    "user": {
      "login": "lee-dohm",
      "contributionsCollection": {
        "startedAt": "2018-05-20T07:00:00Z",
        "endedAt": "2019-05-24T06:59:59Z",
        "contributionYears": [
          2019,
          2018,
          2017,
          2016,
          2015,
          2014,
          2013,
          2012,
          2011
        ],
        "earliestRestrictedContributionDate": null,
        "hasAnyRestrictedContributions": false,
        "hasAnyContributions": true,
        "restrictedContributionsCount": 0
      }
    }
  }
}
</code></pre>
<p>But if I execute the same query but against your user ID then I receive:</p>
<pre><code>{
  "data": {
    "user": {
      "login": "fanny",
      "contributionsCollection": {
        "startedAt": "2018-05-20T07:00:00Z",
        "endedAt": "2019-05-24T06:59:59Z",
        "contributionYears": [
          2019,
          2018,
          2017,
          2016,
          2015
        ],
        "earliestRestrictedContributionDate": "2018-05-25",
        "hasAnyRestrictedContributions": true,
        "hasAnyContributions": true,
        "restrictedContributionsCount": 218
      }
    }
  }
}
</code></pre>
<p>The reverse would be true if you executed the two queries. This is the case because your own contributions are not restricted  <strong>for you</strong> , but they are for others. (This is made a bit more clear in the documentation for <code>restrictedContributionsCount</code>.)</p>
<p>I hope that helps!</p>
